subprojects/gradle/impact/src/main/kotlin/com/avito/impact/plugin/ImpactMetricsSender.kt

```diff
@@ -0,0 +1,48 @@
+package com.avito.impact.plugin
+
+import com.avito.android.sentry.Environment
+import com.avito.android.sentry.EnvironmentInfo
+import com.avito.android.stats.GaugeMetric
+import com.avito.android.stats.StatsDSender
+import com.avito.android.stats.graphiteSeries
+import com.avito.impact.ConfigurationType
+import com.avito.impact.ModifiedProjectsFinder
+import com.avito.impact.configuration.internalModule
+
+class ImpactMetricsSender(
+    private val statsDSender: StatsDSender,
+    private val environmentInfo: EnvironmentInfo
+) {
+
+    init {
+        require(environmentInfo.environment is Environment.CI) {
+            "ImpactMetricsSender should run only in CI environment"
+        }
+    }
+
+    private val prefix by lazy {
+        val envName = environmentInfo.environment.publicName
+        val node = environmentInfo.node?.take(32) ?: "_"
+        val buildId = requireNotNull(environmentInfo.teamcityBuildId()) {
+            "ImpactMetricsSender should run only if teamcityBuildInfo available"
+        }
+        graphiteSeries(envName, node, buildId)
+    }
+
+    fun sendModifiedProjectCounters(modifiedProjectsFinder: ModifiedProjectsFinder) {
```

Comment:
Could we separate transport and `counter`?
I want to see `Sender` that accepts two digits to the `send` function

Reply:
it was like that before last change;
It's not like `StatdSender` it's already extracted;
this class only contains logic about building and sending metrics, don't see any value in separation here

Reply:
API looks strange. The sender accepts `modifiedProjectsFinder` why does it know about how we count modified projects?